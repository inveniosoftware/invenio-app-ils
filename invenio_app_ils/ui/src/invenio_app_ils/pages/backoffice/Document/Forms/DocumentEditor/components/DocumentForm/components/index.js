export { TagsField } from './TagsField';
export { AlternativeAbstracts } from './AlternativeAbstracts';
export { TableOfContent } from './TableOfContent';
export { UrlsField } from './UrlsField';
