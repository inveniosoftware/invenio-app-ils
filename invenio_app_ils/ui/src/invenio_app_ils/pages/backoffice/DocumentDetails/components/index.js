export { DocumentDetails } from './DocumentDetails';
export { DocumentMetadata } from './DocumentMetadata';
export { DocumentPendingLoans } from './DocumentPendingLoans';
export { DocumentAccessRestrictions } from './DocumentAccessRestrictions';
