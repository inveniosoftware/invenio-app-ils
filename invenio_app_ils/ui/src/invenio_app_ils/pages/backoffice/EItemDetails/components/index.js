export { EItemDetails } from './EItemDetails';
export { EItemMetadata } from './EItemMetadata';
