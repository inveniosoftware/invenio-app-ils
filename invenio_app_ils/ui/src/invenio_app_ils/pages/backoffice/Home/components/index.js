export { LoansCard } from './widgets/LoansCard';
export { DocumentsCard } from './widgets/DocumentsCard';
export { ACQRequestsCard } from './widgets/ACQRequestsCard';
export { ILLCard } from './widgets/ILLCard';
export { OverbookedDocumentsList } from './lists/OverbookedDocumentsList';
export { OverdueLoansList } from './lists/OverdueLoansList';
export { IdleLoansList } from './lists/IdleLoansList';
export { RenewedLoansList } from './lists/RenewedLoansList';
