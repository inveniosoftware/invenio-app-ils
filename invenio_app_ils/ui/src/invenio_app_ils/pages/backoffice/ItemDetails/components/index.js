export { ItemDetails } from './ItemDetails';
export { ItemMetadata } from './ItemMetadata';
export { ItemPastLoans } from './ItemPastLoans';
