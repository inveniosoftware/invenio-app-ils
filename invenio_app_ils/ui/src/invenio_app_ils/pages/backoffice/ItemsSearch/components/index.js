export { ResultsList } from './ResultsList';
export { SearchBar } from './SearchBar';
