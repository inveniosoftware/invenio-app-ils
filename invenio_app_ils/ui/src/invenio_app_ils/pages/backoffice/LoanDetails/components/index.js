export { LoanActions } from './LoanActions';
export { LoanDetails } from './LoanDetails';
export { LoanMetadata } from './LoanMetadata';
export { AvailableItems } from './AvailableItems';
