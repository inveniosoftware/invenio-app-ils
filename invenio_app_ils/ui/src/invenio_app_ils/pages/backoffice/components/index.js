export { Header } from './Header';
export { MetadataTable } from './MetadataTable';
export { DeleteRecordModal } from './DeleteRecordModal';
export {
  ExportReactSearchKitResults,
  ExportSearchResults,
} from './ExportSearchResults';
export { OverdueLoanSendMailModal } from './OverdueLoanSendMailModal';
export { Sidebar } from './Sidebar';
