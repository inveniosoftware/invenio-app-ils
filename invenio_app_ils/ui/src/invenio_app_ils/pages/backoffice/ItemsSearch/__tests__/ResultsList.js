import React from 'react';
import omit from 'lodash/omit';
import { mount } from 'enzyme';
import { Settings } from 'luxon';
import { fromISO, toISO } from '../../../../common/api/date';
import { ResultsList } from '../components';
import { formatter } from '../../../../common/components/ResultsTable/formatters';

Settings.defaultZoneName = 'utc';

describe('ItemsSearch ResultsList tests', () => {
  const stringDate = fromISO('2018-01-01T11:05:00+01:00');

  const results = [
    {
      id: 987,
      created: toISO(stringDate),
      metadata: {
        item_pid: 987,
        barcode: '9865745223',
        document_pid: 1342,
        document: {
          document_pid: 1342,
        },
        status: 'IN_BINDING',
        internal_location_pid: 1,
        internal_location: {
          name: 'A library',
          internal_location_pid: 1,
        },
      },
    },
  ];

  let component;
  afterEach(() => {
    component.unmount();
  });

  it('should not render when empty results', () => {
    component = mount(
      <ResultsList results={[]} viewDetailsClickHandler={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a list of results', () => {
    component = mount(
      <ResultsList results={results} viewDetailsClickHandler={() => {}} />
    );
    expect(component).toMatchSnapshot();
    const firstResult = results[0];
    const resultRows = component
      .find('TableRow')
      .filterWhere(
        element => element.prop('data-test') === firstResult.metadata.item_pid
      );
    expect(resultRows).toHaveLength(1);

    const mappedStatusElements = resultRows
      .find('TableCell')
      .filterWhere(
        element =>
          element.prop('data-test') ===
          'Status-' + firstResult.metadata.item_pid
      );
    expect(mappedStatusElements).toHaveLength(1);

    expect(mappedStatusElements.text()).toEqual(firstResult.metadata.status);
  });

  it('should call click handler on view details click', () => {
    const mockedClickHandler = jest.fn();
    component = mount(
      <ResultsList
        results={results}
        viewDetailsClickHandler={mockedClickHandler}
      />
    );
    const firstId = results[0].metadata.item_pid;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('button');
    button.simulate('click');
    const expected = omit(formatter.item.toTable(results[0]), [
      'Created',
      'Updated',
    ]);
    expect(mockedClickHandler).toHaveBeenCalledWith(expected);
  });
});
