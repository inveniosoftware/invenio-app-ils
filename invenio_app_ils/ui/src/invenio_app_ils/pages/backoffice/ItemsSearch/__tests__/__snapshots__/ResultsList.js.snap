// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsSearch ResultsList tests should not render when empty results 1`] = `
<ResultsList
  results={Array []}
  viewDetailsClickHandler={[Function]}
>
  <ResultsTable
    currentPage={1}
    fixed={false}
    headerActionClickHandler={null}
    headerActionComponent={
      <div>
        <NewButton
          clickHandler={[Function]}
          disabled={false}
          fluid={false}
          text="New item"
        />
        <ExportReactSearchKitResults
          exportBaseUrl="https://127.0.0.1:5000/api/items/"
        />
      </div>
    }
    name="items"
    paginationComponent={null}
    renderSegment={true}
    rowActionClickHandler={[Function]}
    rows={Array []}
    seeAllComponent={null}
    showAllResults={false}
    showMaxRows={10}
    singleLine={false}
    subtitle=""
    title=""
  >
    <Segment>
      <div
        className="ui segment"
      >
        <Grid>
          <div
            className="ui grid"
          >
            <GridRow>
              <div
                className="row"
              >
                <GridColumn
                  verticalAlign="middle"
                  width={8}
                >
                  <div
                    className="middle aligned eight wide column"
                  />
                </GridColumn>
                <GridColumn
                  textAlign="right"
                  width={8}
                >
                  <div
                    className="right aligned eight wide column"
                  >
                    <div>
                      <NewButton
                        clickHandler={[Function]}
                        disabled={false}
                        fluid={false}
                        text="New item"
                      >
                        <Button
                          as="button"
                          icon={true}
                          labelPosition="left"
                          onClick={[Function]}
                          positive={true}
                          size="small"
                        >
                          <Ref
                            innerRef={
                              Object {
                                "current": <button
                                  class="ui small icon positive left labeled button"
                                >
                                  <i
                                    aria-hidden="true"
                                    class="plus icon"
                                  />
                                  New item
                                </button>,
                              }
                            }
                          >
                            <RefFindNode
                              innerRef={
                                Object {
                                  "current": <button
                                    class="ui small icon positive left labeled button"
                                  >
                                    <i
                                      aria-hidden="true"
                                      class="plus icon"
                                    />
                                    New item
                                  </button>,
                                }
                              }
                            >
                              <button
                                className="ui small icon positive left labeled button"
                                onClick={[Function]}
                              >
                                <Icon
                                  as="i"
                                  name="plus"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="plus icon"
                                    onClick={[Function]}
                                  />
                                </Icon>
                                New item
                              </button>
                            </RefFindNode>
                          </Ref>
                        </Button>
                      </NewButton>
                      <ExportReactSearchKitResults
                        exportBaseUrl="https://127.0.0.1:5000/api/items/"
                      />
                    </div>
                  </div>
                </GridColumn>
              </div>
            </GridRow>
          </div>
        </Grid>
        <Message
          data-test="no-results"
        >
          <div
            className="ui message"
            data-test="no-results"
          >
            There are no 
            items
            .
          </div>
        </Message>
      </div>
    </Segment>
  </ResultsTable>
</ResultsList>
`;

exports[`ItemsSearch ResultsList tests should render a list of results 1`] = `
<ResultsList
  results={
    Array [
      Object {
        "created": "2018-01-01T10:05:00.000Z",
        "id": 987,
        "metadata": Object {
          "barcode": "9865745223",
          "document": Object {
            "pid": 1342,
          },
          "document_pid": 1342,
          "internal_location": Object {
            "name": "A library",
            "pid": 1,
          },
          "internal_location_pid": 1,
          "pid": 987,
          "status": "IN_BINDING",
        },
      },
    ]
  }
  viewDetailsClickHandler={[Function]}
>
  <ResultsTable
    currentPage={1}
    fixed={false}
    headerActionClickHandler={null}
    headerActionComponent={
      <div>
        <NewButton
          clickHandler={[Function]}
          disabled={false}
          fluid={false}
          text="New item"
        />
        <ExportReactSearchKitResults
          exportBaseUrl="https://127.0.0.1:5000/api/items/"
        />
      </div>
    }
    name="items"
    paginationComponent={null}
    renderSegment={true}
    rowActionClickHandler={[Function]}
    rows={
      Array [
        Object {
          "Barcode": "9865745223",
          "Circulation status": null,
          "Document ID": 1342,
          "ID": 987,
          "Internal location": "A library",
          "Location": null,
          "Medium": undefined,
          "Shelf": undefined,
          "Status": "IN_BINDING",
        },
      ]
    }
    seeAllComponent={null}
    showAllResults={false}
    showMaxRows={10}
    singleLine={false}
    subtitle=""
    title=""
  >
    <Segment>
      <div
        className="ui segment"
      >
        <Grid>
          <div
            className="ui grid"
          >
            <GridRow>
              <div
                className="row"
              >
                <GridColumn
                  verticalAlign="middle"
                  width={8}
                >
                  <div
                    className="middle aligned eight wide column"
                  />
                </GridColumn>
                <GridColumn
                  textAlign="right"
                  width={8}
                >
                  <div
                    className="right aligned eight wide column"
                  >
                    <div>
                      <NewButton
                        clickHandler={[Function]}
                        disabled={false}
                        fluid={false}
                        text="New item"
                      >
                        <Button
                          as="button"
                          icon={true}
                          labelPosition="left"
                          onClick={[Function]}
                          positive={true}
                          size="small"
                        >
                          <Ref
                            innerRef={
                              Object {
                                "current": <button
                                  class="ui small icon positive left labeled button"
                                >
                                  <i
                                    aria-hidden="true"
                                    class="plus icon"
                                  />
                                  New item
                                </button>,
                              }
                            }
                          >
                            <RefFindNode
                              innerRef={
                                Object {
                                  "current": <button
                                    class="ui small icon positive left labeled button"
                                  >
                                    <i
                                      aria-hidden="true"
                                      class="plus icon"
                                    />
                                    New item
                                  </button>,
                                }
                              }
                            >
                              <button
                                className="ui small icon positive left labeled button"
                                onClick={[Function]}
                              >
                                <Icon
                                  as="i"
                                  name="plus"
                                >
                                  <i
                                    aria-hidden="true"
                                    className="plus icon"
                                    onClick={[Function]}
                                  />
                                </Icon>
                                New item
                              </button>
                            </RefFindNode>
                          </Ref>
                        </Button>
                      </NewButton>
                      <ExportReactSearchKitResults
                        exportBaseUrl="https://127.0.0.1:5000/api/items/"
                      />
                    </div>
                  </div>
                </GridColumn>
              </div>
            </GridRow>
          </div>
        </Grid>
        <Table
          as="table"
          selectable={true}
          striped={true}
        >
          <table
            className="ui selectable striped table"
          >
            <ResultsTableHeader
              columns={
                Array [
                  "ID",
                  "Barcode",
                  "Document ID",
                  "Status",
                  "Medium",
                  "Circulation status",
                  "Internal location",
                  "Location",
                  "Shelf",
                ]
              }
              withRowAction={true}
            >
              <TableHeader
                as="thead"
              >
                <thead
                  className=""
                >
                  <TableRow
                    as="tr"
                    cellAs="td"
                    data-test="header"
                  >
                    <tr
                      className=""
                      data-test="header"
                    >
                      <TableHeaderCell
                        as="th"
                        collapsing={true}
                        width={2}
                      >
                        <TableCell
                          as="th"
                          className=""
                          collapsing={true}
                          width={2}
                        >
                          <th
                            className="collapsing two wide"
                          />
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="ID"
                        width={2}
                      >
                        <TableCell
                          as="th"
                          className=""
                          width={2}
                        >
                          <th
                            className="two wide"
                          >
                            ID
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Barcode"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Barcode
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Document ID"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Document ID
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Status"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Status
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Medium"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Medium
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Circulation status"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Circulation status
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Internal location"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Internal location
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Location"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Location
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                      <TableHeaderCell
                        as="th"
                        key="Shelf"
                      >
                        <TableCell
                          as="th"
                          className=""
                        >
                          <th
                            className=""
                          >
                            Shelf
                          </th>
                        </TableCell>
                      </TableHeaderCell>
                    </tr>
                  </TableRow>
                </thead>
              </TableHeader>
            </ResultsTableHeader>
            <ResultsTableBody
              columns={
                Array [
                  "ID",
                  "Barcode",
                  "Document ID",
                  "Status",
                  "Medium",
                  "Circulation status",
                  "Internal location",
                  "Location",
                  "Shelf",
                ]
              }
              rowActionClickHandler={[Function]}
              rows={
                Array [
                  Object {
                    "Barcode": "9865745223",
                    "Circulation status": null,
                    "Document ID": 1342,
                    "ID": 987,
                    "Internal location": "A library",
                    "Location": null,
                    "Medium": undefined,
                    "Shelf": undefined,
                    "Status": "IN_BINDING",
                  },
                ]
              }
            >
              <TableBody
                as="tbody"
              >
                <tbody
                  className=""
                >
                  <TableRow
                    as="tr"
                    cellAs="td"
                    data-test={987}
                    key="987"
                  >
                    <tr
                      className=""
                      data-test={987}
                    >
                      <TableCell
                        as="td"
                        textAlign="center"
                      >
                        <td
                          className="center aligned"
                        >
                          <Button
                            as="button"
                            circular={true}
                            compact={true}
                            data-test="btn-view-details-987"
                            icon="eye"
                            onClick={[Function]}
                          >
                            <Ref
                              innerRef={
                                Object {
                                  "current": <button
                                    class="ui circular compact icon button"
                                    data-test="btn-view-details-987"
                                  >
                                    <i
                                      aria-hidden="true"
                                      class="eye icon"
                                    />
                                  </button>,
                                }
                              }
                            >
                              <RefFindNode
                                innerRef={
                                  Object {
                                    "current": <button
                                      class="ui circular compact icon button"
                                      data-test="btn-view-details-987"
                                    >
                                      <i
                                        aria-hidden="true"
                                        class="eye icon"
                                      />
                                    </button>,
                                  }
                                }
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-987"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                      onClick={[Function]}
                                    />
                                  </Icon>
                                </button>
                              </RefFindNode>
                            </Ref>
                          </Button>
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="ID-987"
                        key="0-987"
                      >
                        <td
                          className=""
                          data-test="ID-987"
                        >
                          987
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Barcode-987"
                        key="1-987"
                      >
                        <td
                          className=""
                          data-test="Barcode-987"
                        >
                          9865745223
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Document ID-987"
                        key="2-987"
                      >
                        <td
                          className=""
                          data-test="Document ID-987"
                        >
                          1342
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Status-987"
                        key="3-987"
                      >
                        <td
                          className=""
                          data-test="Status-987"
                        >
                          IN_BINDING
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Medium-987"
                        key="4-987"
                      >
                        <td
                          className=""
                          data-test="Medium-987"
                        >
                          -
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Circulation status-987"
                        key="5-987"
                      >
                        <td
                          className=""
                          data-test="Circulation status-987"
                        >
                          -
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Internal location-987"
                        key="6-987"
                      >
                        <td
                          className=""
                          data-test="Internal location-987"
                        >
                          A library
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Location-987"
                        key="7-987"
                      >
                        <td
                          className=""
                          data-test="Location-987"
                        >
                          -
                        </td>
                      </TableCell>
                      <TableCell
                        as="td"
                        data-test="Shelf-987"
                        key="8-987"
                      >
                        <td
                          className=""
                          data-test="Shelf-987"
                        >
                          -
                        </td>
                      </TableCell>
                    </tr>
                  </TableRow>
                </tbody>
              </TableBody>
            </ResultsTableBody>
            <ResultsTableFooter
              allRowsNumber={1}
              columnsNumber={9}
              currentPage={1}
              paginationComponent={null}
              seeAllComponent={null}
              showAllResults={false}
              showMaxRows={10}
            />
          </table>
        </Table>
      </div>
    </Segment>
  </ResultsTable>
</ResultsList>
`;
